### PR TITLE
Maintenance/upgrade api

### PR DESCRIPTION
**Pull request by millii19**

### CODE REVIEW

**Comment by absurd-turtle** [comment]
@millii19 What is missing here?

File: api/schema/schema.graphql [hunk 1]
```diff
@@ -117,17 +113,14 @@ type Mutation {
...
 
-type LoginResponse {
-  success: Boolean!
-  message: String
-  token: String
-}
-*/
-module.exports = schema
\ No newline at end of file
+# type Mutation {
```
Comment (absurd-turtle) [inline api/schema/schema.graphql:118]: remove dead code

File: api/src/datasources/spotify/index.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
+// import { DataSource } from 'apollo-datasource';
```
Comment (absurd-turtle) [inline api/src/datasources/spotify/index.ts:1]: remove dead code

File: api/src/directives/RequireAuthDirective.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+            const { requires } = authDirective
+            if (requires) {
+            const { resolve = defaultFieldResolver } = fieldConfig
+            fieldConfig.resolve = function (source, args, context, info) {
+                if (context.user) {
+                    if (!context.user.roles.includes(requires)) {
+                        throw new AuthorizationError()
+                    } else {
+                        return resolve.apply(this, args)
+                        //return resolve(source, args, context, info)
```
Comment (absurd-turtle) [inline api/src/directives/RequireAuthDirective.ts:20]: remove dead code

File: api/src/helpers/formatError.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import { unwrapResolverError } from '@apollo/server/errors';
+import { GraphQLFormattedError } from 'graphql';
+import { isInstance } from 'apollo-errors';
+import { AuthenticationError, NotFoundError } from '../errors/errors.js';
+
+function myFormatError(formattedError: GraphQLFormattedError, error: Error): GraphQLFormattedError {
+  // if (unwrapResolverError(error) instanceof CustomDBError) {
```
Comment (absurd-turtle) [inline api/src/helpers/formatError.ts:7]: .

File: api/src/helpers/formatError.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  //   return { message: 'Internal server error' };
+  // }
+  if (error.message.startsWith('DataLoader must be constructed')) return new NotFoundError('Id Not Found')
+  if (error.message.startsWith('Context creation failed: ')) {
+    error.message = error.message.replace('Context creation failed: ', '')
+  }
+
+  if (error.message.startsWith('AuthenticationError: ')) return new AuthenticationError({ data: { unexpected: error.message.replace('AuthenticationError: ', '') } })
+
+  // if (isInstance(error) && !error.internalData.skiplog) {
```
Comment (absurd-turtle) [inline api/src/helpers/formatError.ts:17]: .

File: api/src/index.ts [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+schema = authDirectiveTransformer(schema, "auth")
+
+
+const server = new ApolloServer({
+  schema: schema,
+  formatError: helpers.formatError,
+  includeStacktraceInErrorResponses: process.env.NODE_ENV === 'development',
+  introspection: process.env.NODE_ENV === 'development',
+  // context: helpers.applyMiddleware([
```
Comment (absurd-turtle) [inline api/src/index.ts:37]: .

**Review by absurd-turtle — CHANGES_REQUESTED** [review]
remove dead code and fix tests otherwise LGTM
Nice job :)

File: api/src/helpers/formatError.ts [hunk 7]
```diff
@@ -1,28 +1,12 @@
...
 function myFormatError(formattedError: GraphQLFormattedError, error: Error): GraphQLFormattedError {
-  // if (unwrapResolverError(error) instanceof CustomDBError) {
-  //   return { message: 'Internal server error' };
-  // }
   if (error.message.startsWith('DataLoader must be constructed')) return new NotFoundError('Id Not Found')
   if (error.message.startsWith('Context creation failed: ')) {
     error.message = error.message.replace('Context creation failed: ', '')
   }
 
-  if (error.message.startsWith('AuthenticationError: ')) return new AuthenticationError({ data: { unexpected: error.message.replace('AuthenticationError: ', '') } })
```
Comment (absurd-turtle) [inline api/src/helpers/formatError.ts]: was this removed intentionally?

**Review by absurd-turtle — APPROVED** [review]
apart from one minor comment it looks good :)

File: api/src/helpers/formatError.ts (hunk 7 above)
Comment (millii19) [reply api/src/helpers/formatError.ts]: yes, I don't believe there should be any more Authentication Errors

Comment (absurd-turtle) [reply api/src/helpers/formatError.ts]: Ah ok, yeah makes sense

**Review by absurd-turtle — APPROVED** [review]

**Pull request merged**